### PR TITLE
API module

### PR DESCRIPTION
**Pull request by brentwatson**
- Created Added `api` module.
- API and Retrofit config setup.
- Created `terms` endpoint implementation.
- Added clone-style constructor to android Terms DTO.
- Updated README with API usage instructions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,24 +4,85 @@
...
+
+### Using Gradle / Maven
+If you are using Gradle or Maven, do the following:
+
+Add jitpack.io as a repository.  [Instructions](https://jitpack.io/docs/#how-to).
+
+To use the latest build, include 1 or more of the following in your `build.gradle` file: 
+`compile 'com.github.percolate.percolate-java-sdk:core:master-SNAPSHOT'`
+`compile 'com.github.percolate.percolate-java-sdk:network:master-SNAPSHOT'`
+`compile 'com.github.percolate.percolate-java-sdk:android:master-SNAPSHOT'`
```
Comment (msya) [inline README.md]: These three dependencies are rendered in one line in the README. Add breaks between each line?

File: README.md [hunk 2]
```diff
@@ -4,24 +4,85 @@
...
+   
+
+### Using Gradle / Maven
+If you are using Gradle or Maven, do the following:
+
+Add jitpack.io as a repository.  [Instructions](https://jitpack.io/docs/#how-to).
+
+To use the latest build, include 1 or more of the following in your `build.gradle` file: 
+`compile 'com.github.percolate.percolate-java-sdk:core:master-SNAPSHOT'`
+`compile 'com.github.percolate.percolate-java-sdk:network:master-SNAPSHOT'`
```
Comment (msya) [inline README.md]: Should this be called compile 'com.github.percolate.percolate-java-sdk:api:master-SNAPSHOT'`?
This dependency didn't resolve for network.

File: README.md (hunk 1 above)
Comment (brentwatson) [inline README.md]: :heavy_check_mark:

File: README.md (hunk 2 above)
Comment (brentwatson) [inline README.md]: :heavy_check_mark: Yes.  I renamed from `network` -> `api` and forgot to update this ref

File: api/src/main/java/com/percolate/sdk/api/PercolateApi.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    public static final String API_V3_PATH = "api/v3";
+    public static final String API_V4_PATH = "api/v4";
+    public static final String API_V5_PATH = "api/v5";
+
+    /**
+     * Create percolate API instance that accesses {@code percolate.com} endpoint.
+     *
+     * @param apiKey API key.
+     */
+    public PercolateApi(String apiKey) {
```
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/PercolateApi.java]: Should this have a NonNull annotation?

Comment (brentwatson) [inline api/src/main/java/com/percolate/sdk/api/PercolateApi.java]: :heavy_check_mark: Done in next PR.

File: README.md [hunk 4]
```diff
@@ -4,24 +4,87 @@
...
+Terms terms = call.execute().body();
+```
+
+**Asynchronously**
+```java
+call.enqueue(new Callback<Terms>() {
+
+    @Override
+    public void onResponse(Call<Terms> call, Response<Terms> response) {
+        Terms terms = response.body();
```
Comment (msya) [inline README.md]: I found if a missing parameter is sent, then `response.body()` is null.

File: api/src/main/java/com/percolate/sdk/api/request/terms/TermsParams.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Parameters for Terms request.
+ */
+@SuppressWarnings("unused")
+public class TermsParams {
+    private Map<String, Object> params = new HashMap<>();
+
+    public TermsParams() {
+        params.put("limit", "1000");
```
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/request/terms/TermsParams.java]: Should there be an option for the user to set a limit?

File: api/src/main/java/com/percolate/sdk/api/utils/RetrofitApiFactory.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+import com.percolate.sdk.api.PercolateApi;
+import retrofit2.Retrofit;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Creates a Retrofit Service interface and caches it.
+ */
+public class RetrofitApiFactory {
```
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/utils/RetrofitApiFactory.java]: Should this class be private?

File: README.md (hunk 4 above)
Comment (brentwatson) [inline README.md]: Added `Error` DTO's for convenience, but the Retrofit way of dealing with errors is to check `response.isSuccess()` and if false, use `response.errorBody()`.

File: api/src/main/java/com/percolate/sdk/api/request/terms/TermsParams.java (hunk 5 above)
Comment (brentwatson) [inline api/src/main/java/com/percolate/sdk/api/request/terms/TermsParams.java]: Here I was showing off how we'd set default values.  Other `XXXXParams` objects I've put together have setters for pagination fields.  Example: 
https://github.com/percolate/percolate-java-sdk/blob/approvals_api_requests/api/src/main/java/com/percolate/sdk/api/request/approvals/pools/ApprovalPoolsParams.java#L31-L54

For the terms endpoint I think always getting everything is OK.

File: api/src/main/java/com/percolate/sdk/api/utils/RetrofitApiFactory.java (hunk 6 above)
Comment (brentwatson) [inline api/src/main/java/com/percolate/sdk/api/utils/RetrofitApiFactory.java]: I wish java had better scoping rules.  :(  Having this private to the Lib would be nice, but it's used by XXXXRequest.java objects, which will each have their own package.  So this must remain public.

File: README.md (hunk 4 above)
Comment (msya) [inline README.md]: :+1:

File: api/src/main/java/com/percolate/sdk/api/request/terms/TermsParams.java (hunk 5 above)
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/request/terms/TermsParams.java]: :+1:

File: api/src/main/java/com/percolate/sdk/api/utils/RetrofitApiFactory.java (hunk 6 above)
Comment (msya) [inline api/src/main/java/com/percolate/sdk/api/utils/RetrofitApiFactory.java]: :+1:

**Pull request merged**